packages/frontend/src/components/NavBar/UserCredentialsSwitcher.tsx

```diff
@@ -0,0 +1,123 @@
+import { ActionIcon, MantineProvider, Menu } from '@mantine/core';
+import { IconCheck, IconDatabaseCog, IconPlus } from '@tabler/icons-react';
+import { useMemo, useState } from 'react';
+import { useActiveProjectUuid } from '../../hooks/useActiveProject';
+import { useProjects } from '../../hooks/useProjects';
+import {
+    useProjectUserWarehouseCredentialsPreference,
+    useProjectUserWarehouseCredentialsPreferenceMutation,
+} from '../../hooks/userWarehouseCredentials/useProjectUserWarehouseCredentialsPreference';
+import { useUserWarehouseCredentials } from '../../hooks/userWarehouseCredentials/useUserWarehouseCredentials';
+import MantineIcon from '../common/MantineIcon';
+import { CreateCredentialsModal } from '../UserSettings/MyWarehouseConnectionsPanel/CreateCredentialsModal';
+
+const UserCredentialsSwitcher = () => {
+    const [isCreatingCredentials, setIsCreatingCredentials] = useState(false);
+    const {
+        isInitialLoading: isLoadingCredentials,
+        data: userWarehouseCredentials,
+    } = useUserWarehouseCredentials();
+    const { isInitialLoading: isLoadingProjects, data: projects } =
+        useProjects();
+    const { isLoading: isLoadingActiveProjectUuid, activeProjectUuid } =
+        useActiveProjectUuid();
+    const { data: preferredCredentials } =
+        useProjectUserWarehouseCredentialsPreference(activeProjectUuid);
+    const { mutate } = useProjectUserWarehouseCredentialsPreferenceMutation();
+
+    const activeProject = useMemo(() => {
```

Comment:
wonder if `useActiveProject` could be used here 
(non-blocking)

Reply:
`useActiveProject` returns a uuid (string) from localstorage. 
I'm using `useActiveProjectUuid` which covers that and more scenarios. 
I agree these hooks are confusing and none returns a project object :/